File: ui/app/common/pnc-client/resources/GenericSetting.js
/*
 * JBoss, Home of Professional Open Source.
 * Copyright 2014-2020 Red Hat, Inc., and individual contributors
 * as indicated by the @author tags.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
(function () {
  'use strict';

  var module = angular.module('pnc.common.pnc-client.resources');

  module.factory('GenericSetting', [
    '$http',
    'restConfig',
    function ($http, restConfig) {

      var ENDPOINT = restConfig.getPncRestUrl() + '/generic-setting';

      function inMaintenanceMode() {
        return $http.get(ENDPOINT + '/in-maintenance-mode');
      }
      function activateMaintenanceMode(reason) {
        return $http.post(ENDPOINT + '/activate-maintenance-mode', reason);
      }
      function deactivateMaintenanceMode() {
        return $http.post(ENDPOINT + '/deactivate-maintenance-mode');
      }
      function getAnnouncementBanner() {
        return $http.get(ENDPOINT + '/announcement-banner');
      }
      function setAnnouncementBanner(message) {
        return $http.post(ENDPOINT + '/announcement-banner', message);
      }

      return {
        inMaintenanceMode, activateMaintenanceMode, deactivateMaintenanceMode, getAnnouncementBanner, setAnnouncementBanner
      };

    }

  ]);

})();

File: model/src/main/java/org/jboss/pnc/model/GenericSetting.java
/**
 * JBoss, Home of Professional Open Source.
 * Copyright 2014-2020 Red Hat, Inc., and individual contributors
 * as indicated by the @author tags.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.jboss.pnc.model;

import java.util.Objects;
import org.hibernate.annotations.Type;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Index;
import javax.persistence.Lob;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;

@Entity
@Table(indexes = { @Index(name = "idx_key", columnList = "key") })
public class GenericSetting implements GenericEntity<Integer> {

    public static final String SEQUENCE_NAME = "generic_setting_id_seq";

    @Id
    @SequenceGenerator(name = SEQUENCE_NAME, sequenceName = SEQUENCE_NAME, allocationSize = 1, initialValue = 100)
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = SEQUENCE_NAME)
    private Integer id;

    @NotNull
    @Column(unique = true, updatable = false)
    private String key;

    @NotNull
    @Lob
    @Type(type = "org.hibernate.type.TextType")
    private String value;

    @Override
    public Integer getId() {
        return id;
    }

    @Override
    public void setId(Integer id) {
        this.id = id;
    }

    public String getKey() {
        return key;
    }

    public void setKey(String key) {
        this.key = key;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o)
            return true;
        if (!(o instanceof GenericSetting))
            return false;
        GenericSetting that = (GenericSetting) o;
        return key.equals(that.getKey());
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(key);
    }
}
